typing/parmatch.ml

```diff
@@ -2363,18 +2355,31 @@ let rec matrix_stable_vars rs = match rs with
     Vars (List.fold_left IdSet.union IdSet.empty stables_in_varsets)
 | rs ->
     let rs = simplify_first_col rs in
-    if not (all_coherent (first_column rs)) then All
+    if not (all_coherent (first_column rs))
+    (* If the column is ill-typed but deemed coherent, we might spuriously
+       warn about some variables being unstable.
+       As sad as that might be, the warning can be silenced by splitting the
+       or-pattern...
+    *)
```

Comment:
In the other places checking coherence, the comment discussing what happens for *coherent-but-ill-typed* columns is in the else branch, I don't know why you moved it here.

Reply:
Ok, I moved it back.